github-workflow-roles/main.tf

```diff
@@ -5,25 +5,44 @@
 locals {
   aws_account_id = var.aws_account_id != "" ? var.aws_account_id : data.aws_caller_identity.current.account_id
 
-  role_name        = coalesce(var.role_name, "github-workflow-role-${var.repository_name}")
-  role_description = "Github Workflow Role for ${var.github_organization_name}/${var.repository_name}"
+  role_name        = var.principal_type == "github" ? coalesce(var.role_name, "github-workflow-role-${var.repository_name}") : var.role_name
```

Comment:
Can we shorten up the GitHub workflow role name now that we're using path-based policies?